compiler/rustc_borrowck/src/type_check/mod.rs

```diff
@@ -1057,15 +1057,19 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
     fn check_user_type_annotations(&mut self) {
         debug!(?self.user_type_annotations);
         for user_annotation in self.user_type_annotations {
-            let CanonicalUserTypeAnnotation { span, ref user_ty, inferred_ty } = *user_annotation;
+            let CanonicalUserTypeAnnotation { span, ref user_ty, inferred_ty, variance } =
+                *user_annotation;
             let inferred_ty = self.normalize(inferred_ty, Locations::All(span));
             let annotation = self.instantiate_canonical_with_fresh_inference_vars(span, user_ty);
             match annotation {
                 UserType::Ty(mut ty) => {
```

Comment:
this is probably also broken for `UserType::TypeOf` :thinking: tbh I don't really want variance here